### ISSUE TITLE
metric_definitions and annotation_definitions files get removed

### ISSUE THREAD

**Issue by twarrendewit** [Unity-Technologies/com.unity.perception#470]
I have a unity-perception scene with instance segmentation. The instances get dynamically loaded at the beginning of each iteration.

When I run iterations on the order of 3-10, everything is fine. But when I run in the range 100-1000 iterations, I come back to find no metric_definitions.json or annotation_definitions.json files.

What could be causing this?

**Comment by twarrendewit** [comment]
Unfortunately I can't post the project since it's for a client. I could potentially make a small sample but that would take a good bit of time to reproduce.

However, I can pull out some snippets. I'm placing random objects on a table, and to avoid bias we spawn them randomly. Then I check for collisions, and if there's any, I restart.

```
private GameObject ReadyItem(GameObject item, Vector3 position, float placeDiameterInches)
{
    var gameObject = GameObject.Instantiate(item, position, Quaternion.identity);

    var mesh = gameObject;
    if( gameObject.transform.childCount > 0)
        mesh = gameObject.transform.GetChild(0).gameObject;
    else
        Debug.Log("No child mesh on "+item.name);

    gameObject.name = item.name;

    mesh.name = item.name+"_mesh";

    var collider = mesh.AddComponent<BoxCollider>();
    // collider.convex = true;

    var rigidBody = mesh.AddComponent<Rigidbody>();
    rigidBody.useGravity = false;
    rigidBody.constraints = RigidbodyConstraints.FreezePositionX | RigidbodyConstraints.FreezePositionY | RigidbodyConstraints.FreezePositionZ
        | RigidbodyConstraints.FreezeRotationX | RigidbodyConstraints.FreezeRotationY | RigidbodyConstraints.FreezeRotationZ;

    var labeling = gameObject.AddComponent<Labeling>();
    labeling.labels.Add(item.name);

    var itemCollision = mesh.AddComponent<ItemCollision>();

    var length = (float) _item[item.name]["Item Length (Inches)"];
    var width = (float) _item[item.name]["Item Width (Inches)"];
    var height = (float) _item[item.name]["Item Height (Inches)"];
    var weight = (float) _item[item.name]["Item Weight (Grams)"];

    var itemWeight = mesh.AddComponent<ItemWeight>();
    itemWeight.WeightGrams = weight;

    var placeRenderer = transform.GetComponent<Renderer>();
    var itemRenderer = gameObject.GetComponentInChildren<Renderer>();

    var inchesToUnityRatio = placeDiameterInches / placeRenderer.bounds.size.x;
    var maxMeshDimension = Mathf.Max(itemRenderer.bounds.size.x,itemRenderer.bounds.size.z);
    var maxRealInches = Mathf.Max(length,width);
    var itemInchesToUnityRatio = maxRealInches / maxMeshDimension;
    var unityToRealRatio =  itemInchesToUnityRatio / inchesToUnityRatio;

    gameObject.transform.localScale *= unityToRealRatio;

    var meshMaxFlat = Mathf.Max(itemRenderer.bounds.size.x,itemRenderer.bounds.size.z);
    var itemMaxFlat = Mathf.Max(length,width);

    var placePercentUnity = meshMaxFlat / placeRenderer.bounds.size.x;
    var placePercentReal = itemMaxFlat / placeDiameterInches;

    if( Mathf.Abs(placePercentReal - placePercentUnity) > 0.01 )
        Debug.LogWarning("Item scale mismatch for "+item.name+". Real % = "+placePercentReal+" Unity % = "+placePercentUnity);

    return gameObject;
}

private IEnumerator TryPlatingItems(List<GameObject> itemsToPlace, float placeDiameterInches, float iterationDelaySeconds, bool rotateToFit, FloatParameter offsetParameter, FloatParameter angleParameter, FloatParameter rotationParameter, FloatParameter scaleParameter)
{
    var placeRenderer = transform.GetComponent<Renderer>();

    // Remove items.
    while (transform.childCount > 0)
    {
        Destroy(transform.GetChild(0).gameObject);
        yield return new WaitForSeconds(iterationDelaySeconds);
    }

    // Place each item.
    foreach (var selectedItem in itemsToPlace)
    {
        var item = ReadyItem(selectedItem, transform.position, placeDiameterInches);
        item.transform.SetParent(transform);

        var placeRadius = placeRenderer.bounds.size.x / 2.0f;
        var placeCenter = placeRenderer.bounds.center;

        var renderer = item.GetComponentInChildren<Renderer>();

        var size = Mathf.Max(renderer.bounds.size.x, renderer.bounds.size.z) / 2.0;

        var max = placeRadius - size;
        max *= 1.1f;

        var angle = angleParameter.Sample();
        var offset = (float) (offsetParameter.Sample() * max);

        var height = renderer.bounds.size.y/2.0;

        item.transform.position = placeCenter + GetPosition(angle, offset) + new Vector3(0,(float) height+0.01f,0);

        yield return new WaitForSeconds(iterationDelaySeconds);
    }

    // Randomly scale the items.
    for (var i = 0; i < transform.childCount; i++)
    {
        var scale = scaleParameter.Sample();
        var item = transform.GetChild(i);
        item.transform.localScale *= scale;

        var itemWeight = item.GetComponentInChildren<ItemWeight>();
        itemWeight.WeightGrams *= scale;
    }

    // Randomly rotate the items.
    for (var i = 0; i < transform.childCount; i++)
    {
        var rotation = rotationParameter.Sample();
        var item = transform.GetChild(i);
        item.transform.localRotation = Quaternion.Euler(0, rotation, 0);
        yield return new WaitForSeconds(iterationDelaySeconds);

        var incremented = 1.0f;
 
        // Try rotating until stuff fits.
        if (rotateToFit)
        {
            while (!IsCollisionFree() && incremented < 360.0)
            {
                item.transform.localRotation = Quaternion.Euler(0, rotation + incremented, 0);
                incremented++;
            }
        }
    }

    yield return null;
}
```

**Comment by StevenBorkman** [comment]
Hey all! The Perception team just released 🎉 Perception 1.0 🎉, a major update to the toolset! Along with all of the new major improvements, we have fixed a lot of other bugs that we have discovered along the way, including this one reported by you. Thanks for being an active user.

There are a bunch of cool features to check out such as ray tracing, depth, a new output format SOLO, normals, and more! Here is the full list of changes: [Perception 1.0 Changelog](https://github.com/Unity-Technologies/com.unity.perception/blob/main/com.unity.perception/CHANGELOG.md)
